### PR TITLE
fix: align sampler batch size with decode batch size

### PR DESCRIPTION
**Pull request by rebel-daeyang**
### 🚀 Summary of Changes

Previously, the decoder output logits were sliced before being passed to the sampler. However, performing slicing operations on rbln device tensors triggers unnecessary d2h and h2d transfers, leading to a significant increase in latency.

To resolve this performance issue, this PR removes the logits slicing logic and updates the sampler to accept the full logits, matching the decode batch size.

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] 🚀 Release (`release`)
* [ ] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [x] 🛠 Bug fix (`fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
2. Verify output: `...`
3. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [ ] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

Verified outputs for 1~64 batch size, temperature/top_p/top_k, chunked prefill with llama3.2 model.
Verified outputs for medusa, lora (eagle3 fails as before).
rbln device tensor input does not yet support speculative decoding or LoRA.
Request for Review: Please check for potential issues regarding Speculative Decoding, LoRA, or MoE models.

---

### CODE REVIEW

File: vllm_rbln/v1/worker/rbln_model_runner.py
Comment (rebel-ykchoi) [inline vllm_rbln/v1/worker/rbln_model_runner.py:1956]: Please, remove the code for compiling the sampler every num_tokens in the decode stage as it is no longer necessary.

**Review by rebel-ykchoi — CHANGES_REQUESTED** [review]

**Comment by rebel-daeyang** [comment]
I noticed that applying this PR removes logits slicing for CPU tensor inputs, which leads to a decrease in sampler performance.
I will revisit this once the performance of device tensor inputs surpasses the current CPU tensor input performance.

**Comment by rebel-daeyang** [comment]
Included in #548 and merged. Closing this PR.
